Fix #44 SI-9060 XML 5th edition name characters

ashawley opened the pull request:
Refactored PR from #44 by @som-snytt to only make minimal changes to tokenization code, namely the acceptance of the "MIDDLE DOT" character, #xB7, from the original ticket.

To follow the standard, a colon should be allowed to start a name.  It seems that a colon could always start an XML name -- in addition to a letter or an underscore.  Not sure why it was avoided, but it's probably pretty rare, anyway.

https://www.w3.org/TR/2008/REC-xml-20081126
https://www.w3.org/TR/2006/REC-xml-20060816
https://www.w3.org/TR/2004/REC-xml-20040204
https://www.w3.org/TR/2000/REC-xml-20001006
https://www.w3.org/TR/1998/REC-xml-19980210

som-snytt commented:
xml parsing used to be shared with compiler, or cut & pasted from it.

The compiler has to know when to enter "xml mode".
```
val <: = 42
val `<:` = 42
```

I don't see why you'd want to make something minimally correct with a minimal change.

ashawley commented:
Thanks for the feedback, @som-snytt.

I have added a new issue, #94, for looking further at names starting with a colon.  I added some examples, but please add more so I can understand how the colon should ideally work with XML names.

Thanks again,
Aaron

SethTisue commented:
> I don't see why you'd want to make something minimally correct with a minimal change.

@som-snytt I don't understand what you mean...?

som-snytt commented:
@SethTisue I don't understand why my other PR was cannibalized instead of merged. I think my comment meant, Why take correctness piecemeal instead of whole hog?

ashawley commented:
@som-snytt It's been a while, but I recall that the minimal changes were able to make your unit test you provided pass.   I was attempting to improve the likelihood that a fix would get merged, by making a more minimal change based on your work.  We can see that that didn't work out.

SethTisue commented:
needs rebase because of tests moving around
